### PR TITLE
[Console] ChoiceQuestion must have choices

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #22842
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by javiereguiluz** [comment]
What is the behavior proposed here? If you pass zero choices to a choice question, shouldn't we throw an exception?

**Comment by ro0NL** [comment]
Perhaps easier yes. Wasnt really sure about the default value behavior (without any further options); or a default value being an unknown option.

The first (actually the case from the issue) also crashes, the latter is already validated when the question is optional.

So :+1: for throwing in the constructor, unless im unaware of any side effects :)

**Comment by ro0NL** [comment]
Updated.

**Review by javiereguiluz — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Component/Console/Question/ChoiceQuestion.php [hunk 1]
```diff
@@ -32,6 +32,10 @@ class ChoiceQuestion extends Question
      */
     public function __construct($question, array $choices, $default = null)
     {
+        if (!$choices) {
+            throw new \LogicException('Choice question must have at least 1 choice available.');
```
Comment (SpacePossum) [inline src/Symfony/Component/Console/Question/ChoiceQuestion.php]: maybe `InvalidArgumentException` would be a better fit here?

**Comment by fabpot** [comment]
Thank you @ro0NL.

File: src/Symfony/Component/Console/Question/ChoiceQuestion.php (hunk 1 above)
Comment (nunomaduro) [inline src/Symfony/Component/Console/Question/ChoiceQuestion.php]: @SpacePossum I totally agree with u.

Comment (ro0NL) [inline src/Symfony/Component/Console/Question/ChoiceQuestion.php]: not sure if we really care :) imo. the argument is valid (an array), but it's domain is invalid (empty array). So `DomainException` actually (imo.) but i thought `LogicException` is a bit more common in SF :)

Im fine either way.. as long as it's extends from logic i dont mind :)

Comment (SpacePossum) [inline src/Symfony/Component/Console/Question/ChoiceQuestion.php]: changing it now would be BC break ;) anyway we should be good. If it is a domain level exception I would expect it to be thrown on count < 2 as  one option does not leave anything to choose from either ;)

**Pull request closed without merging**